### PR TITLE
[TTAHUB-1202] Track the order objectives were added to the RTR/create goals

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Add a DB column to track the order objectives were added to the RTR. Sort by that column.

## How to test
Add multiple objectives to a goal on the create goals form. Save. Check that they emerge in the same order.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1202

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/services/goals.js [hunk 1]
```diff
@@ -1012,6 +1013,7 @@ export async function createOrUpdateGoals(goals) {
         await objective.update({
           title,
           status: objectiveStatus,
+          rtrOrder: index + 1,
         }, { individualHooks: true });
```
Comment (AdamAdHocTeam) [inline src/services/goals.js:1017]: Very elegant fix. I would have thought it would need to be on the FE. 😄

**Review by AdamAdHocTeam — COMMENTED** [review]
- Ok create a Goal with three objectives fill out data and save
- Edit and go to the first objective remove the topics and click save (it shouldn't due to validation) 
- Fill in the missing topics and save again
- You will see the incorrect order (also when you edit)

**Note:** I wasn't able to get this result every time so it might be an saving order thing.

**Comment by thewatermethod** [comment]
Made a small change - I am not able to break this any longer (although I wasn't able to consistently do so before either)

**Review by AdamAdHocTeam — APPROVED** [review]
I coudln't break it this time.

**Pull request merged**